### ISSUE TITLE
add config.w32 to aid compilation on windows

### ISSUE THREAD

**Issue by Hamfri** [php-gnupg/php-gnupg#21]
Been trying for days now to compile this library to php_gnu.dll which is almost impossible because it lacks config.w32 and hence I have been forced to try and convert config.m4 to config.w32. I think it would be nice to include this file too so that those on windows are also able to use the extension with ease.

**Comment by KaloyNayve** [comment]
Can any windows expert there help out with this. I really need to compile php_gnu.dll for windows but I am having trouble creating a config.w32 by myself

**Comment by bukka** [comment]
I'm not really a Windows expert but it's not just config.w32 but also getting the gpgme which should be probably part of  [Gpg4win](https://www.gpg4win.org/download.html). I personally don't have an idea how to put it together on Window and how to make those automatic builds on PECL. I'm of course happy to merge a PR if someone is able to do that. Maybe @cmb69 or @weltling might be able to help with that.

**Comment by cmb69** [comment]
Right, the biggest hurdle would be libgpgme; it's available via vcpkg, but only for *nix. The binary Gpg4win distribution contains development files, but is only available for x86; the tarballs they're shipping are apparently build with Msys/MinGW, so can't be used as well. The original gpgme also requires an autotools build environment. It would be great if somebody would add a CMake based build chain, or at least provide VS project files for libgpgme and its dependencies. Then we likely could build and ship binary packages for PECL.

A minimal config.w32 could be something like:
````js
ARG_WITH("gnupg", "gnupg support", "no");

if (PHP_GNUPG != "no") {
	if (CHECK_HEADER_ADD_INCLUDE("gpgme.h", "CFLAGS_GNUPG", PHP_PHP_BUILD + "\\include;" + PHP_EXTRA_INCLUDES) &&
		CHECK_LIB("libgpgme.lib", "gnupg", PHP_GNUPG)
	) {
		EXTENSION("gnupg", "gnupg.c gnupg_keylistiterator.c");
		AC_DEFINE('HAVE_GNUPG', 1);
	} else {
		WARNING("gnupg not enabled; libraries or headers not found");
	}
}
````
Adding large file support for 32bit Windows would likely be quite some additional work.

**Comment by bukka** [comment]
Thanks @cmb69 for checking it. That's very useful. I had a feeling that getting libgpgme build will be the main obstacle.
